src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -134,7 +175,34 @@ public void Dispose()
             }
         }
 
-        private async Task<string> GetLastPollStatusAsync()
+        private async Task MoveToPoisonQueueAsync(ApiHubFile apiHubFile)
+        {
+            CloudQueueClient queueClient = CloudStorageAccount.Parse(this._config.StorageConnectionString).CreateCloudQueueClient();
+
+            var queue = queueClient.GetQueueReference(PoisonQueueName);
+
+            await queue.CreateIfNotExistsAsync();
+
+            var fileStatus = new ApiHubFileInfo
+            {
+                FilePath = apiHubFile.Path,
+                FunctionName = this._functionName,
+                Connection = this._connectionStringSetting
```

Comment:
Not sure, why is this needed? Am I missing something? This is just to let the user know which trigger failed. We are giving them the file path, function name and the connection string name which is something like DROPBOX_dropbox for them to know which connector was the file coming from which failed.